finality-aleph/src/network/session/compatibility.rs

```diff
@@ -19,129 +16,36 @@ type ByteCount = u16;
-    pub fn add_authentication(&mut self, authentication: Authentication<A>) {
-        use PeerAuthentications::*;
-        match self {
-            NewOnly(_) => *self = NewOnly(authentication),
-            LegacyOnly(legacy_authentication) | Both(_, legacy_authentication) => {
-                *self = Both(authentication, legacy_authentication.clone())
-            }
-        }
-    }
-
-    /// Add a legacy authentication, overriding one that might have been here previously.
-    pub fn add_legacy_authentication(&mut self, legacy_authentication: LegacyAuthentication<M>) {
-        use PeerAuthentications::*;
-        match self {
-            LegacyOnly(_) => *self = LegacyOnly(legacy_authentication),
-            NewOnly(authentication) | Both(authentication, _) => {
-                *self = Both(authentication.clone(), legacy_authentication)
-            }
-        }
-    }
-
-    /// The associated address, if any. Can be `None` only for legacy authentications.
-    pub fn maybe_address(&self) -> Option<A> {
-        use PeerAuthentications::*;
-        match self {
-            NewOnly((auth_data, _)) | Both((auth_data, _), _) => Some(auth_data.address()),
-            LegacyOnly((legacy_auth_data, _)) => legacy_auth_data
-                .clone()
-                .try_into()
-                .map(|auth_data: AuthData<A>| auth_data.address())
-                .ok(),
-        }
-    }
-}
-
-/// Legacy messages used for discovery and authentication.
-#[derive(Clone, Debug, PartialEq, Eq, Hash, Encode, Decode)]
-pub enum LegacyDiscoveryMessage<M: Data> {
-    AuthenticationBroadcast(LegacyAuthentication<M>),
-    Authentication(LegacyAuthentication<M>),
-}
-
 #[derive(Clone, Debug, PartialEq, Eq)]
-pub enum VersionedAuthentication<M: Data, A: AddressingInformation> {
+pub enum VersionedAuthentication<A: AddressingInformation> {
     // Most likely from the future.
     Other(Version, Vec<u8>),
-    V1(LegacyDiscoveryMessage<M>),
     V2(Authentication<A>),
 }
 
-impl<M: Data, A: AddressingInformation + TryFrom<Vec<M>> + Into<Vec<M>>>
-    From<PeerAuthentications<M, A>> for Vec<VersionedAuthentication<M, A>>
-{
-    fn from(authentications: PeerAuthentications<M, A>) -> Self {
-        use LegacyDiscoveryMessage::*;
-        use PeerAuthentications::*;
+impl<A: AddressingInformation> From<Authentication<A>> for Vec<VersionedAuthentication<A>> {
+    fn from(authentication: Authentication<A>) -> Self {
         use VersionedAuthentication::*;
-        match authentications {
-            NewOnly(authentication) => vec![V2(authentication)],
-            LegacyOnly(legacy_authentication) => {
-                vec![V1(AuthenticationBroadcast(legacy_authentication))]
-            }
-            Both(authentication, legacy_authentication) => vec![
-                V2(authentication),
-                V1(AuthenticationBroadcast(legacy_authentication)),
-            ],
-        }
+        vec![V2(authentication)]
```

Comment:
The `use` above is a bit of an overkill, just spell the whole thing out in one line. ;p

Reply:
fixed